internal/config/validator.go

```diff
@@ -58,13 +58,65 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Nodes = append(clusterResources.Nodes, list.Items...)
 		}
 	}
+	clusterResources = resetTransientErrorsFields(clusterResources)
 	_, err := For(clusterResources, v.validate)
-	if errors.As(err, &TransientError{}) { // we do not want to make assumption on ordering in webhooks.
-		return nil
-	}
 	return err
 }
 
 func NewValidator(validate Validate) apivalidate.ClusterObjects {
 	return &validator{validate: validate}
 }
+
+// resetting the fields that lead to Transienterror.
+func resetTransientErrorsFields(clusterResources ClusterResources) ClusterResources {
+	for i := range clusterResources.Peers {
+		clusterResources.Peers[i].Spec.BFDProfile = ""
+		clusterResources.Peers[i].Spec.PasswordSecret.Name = ""
+	}
+	for i, bgpAdv := range clusterResources.BGPAdvs {
+		var communities []string
+		for _, community := range bgpAdv.Spec.Communities {
+			if existCommunity(community, clusterResources.Communities) {
+				communities = append(communities, community)
+			} else if !checkInvalidCommunityFormat(community) {
+				communities = append(communities, community)
+			}
```

Comment:
I think this gets into the validation details too much, rather than focusing on "avoiding transient errors". What I mean is what we actually want is to not pass references to other objects (like with the bfdprofiles) without caring about the actual validation. Since it is a bit harder with this field (having multiple formats), I'd say we can check here if the string is a "valid reference" and drop it if so, and the conditions will be replaced by:
```go
	if errs := validation.IsDNS1123Subdomain(community); len(errs) != 0 {
		communities = append(communities, community)
	}
```
(using `"k8s.io/apimachinery/pkg/util/validation"`)
this still makes the assumption that a community value can not be of this format (which k8s object names follow), but I think it's better than the specific community validations.